tools/translation/config.js

```diff
@@ -0,0 +1,178 @@
+    // reshape object for easy access
+    config = {
+      locales: json.locales.data,
+      glaas: {
+        ...g,
+        workflows: {}
+      },
+      sp
+    };
+    json.workflows.data.forEach(w => {
+      config.glaas.workflows[w.name] = w;
+    });
+
+    const location = new URL(window.location.href);
+
+    config.glaas = {
+      ...config.glaas,
+      authorizeURI: '/api/common/sweb/oauth/authorize',
+      redirectURI: encodeURI(`${location.origin}/tools/translation/glaas.html`),
+      accessToken: null,
+      api: {
+        session: {
+          check: {
+            uri: '/api/common/v1.0/checkSession',
+          },
+        },
+      },
+      localeApi: async (locale) => {
+        const workflow = await getWorkflowForLocale(locale);
+        const { product, project, workflowName } = workflow;
+        return {
+          tasks: {
+            create: {
+              uri: `/api/l10n/v1.1/tasks/${product}/${project}/create`,
+              payload: {
+                workflowName: workflowName,
+                contentSource: 'Adhoc'
+              },
+            },
+            get: {
+              baseURI: `/api/l10n/v1.1/tasks/${product}/${project}`,
+            },
+            getAll: {
+              uri: `/api/l10n/v1.1/tasks/${product}/${project}`,
+            },
+            updateStatus: {
+              baseURI: `/api/l10n/v1.1/tasks/${product}/${project}`,
+            },
+            assets: {
+              baseURI: `/api/l10n/v1.1/tasks/${product}/${project}`,
+            },
+          }
+        };
+      }
+    };
+
+    const graphURL = 'https://graph.microsoft.com/v1.0';
+
+    config.sp = {
+      ...config.sp,
+      clientApp: {
+        auth: {
+          clientId: config.sp.clientId,
+          authority: config.sp.authority,
+        },
+      },
+      login: {
+        redirectUri: '/tools/translation/spauth',
+      },
+      api: {
+        url: graphURL,
+        file: {
+          get: {
+            baseURI: `${config.sp.site}/drive/root:${config.sp.rootFolders}`,
+          },
+          download: {
+            baseURI: `${config.sp.site}/drive/items`,
+          },
+          upload: {
+            baseURI: `${config.sp.site}/drive/root:${config.sp.rootFolders}`,
+            method: 'PUT',
+          },
+          createUploadSession: {
+            baseURI: `${config.sp.site}/drive/root:${config.sp.rootFolders}`,
+            method: 'POST',
+            payload: {
+              '@microsoft.graph.conflictBehavior': 'replace',
+            },
+          },
+        },
+        directory: {
+          create: {
+            baseURI: `${config.sp.site}/drive/root:${config.sp.rootFolders}`,
+            method: 'PATCH',
+            payload: {
+              folder: {},
+            },
+          },
+        },
+        batch: {
+          uri: `${graphURL}/$batch`,
+        },
+      },
+    };
+
+    const adminServerURL = 'https://admin.hlx3.page';
```

Comment:
I understand the code is expecting hlx3, but we're not ready in `master`; maybe the PR should be against `main`?
cc @rofe

Reply:
The admin API works also with hlx2 sites for some operations. Since you are migrating to hlx3 anyway, this will not be an issue soon.

Reply:
Yes, migration is ongoing. As long as it doesn't break functionality, it shouldn't be an issue.
cc @rofe